app/screens/search/search.js

```diff
@@ -0,0 +1,573 @@
+import SearchBar from 'app/components/search_bar';
+import SearchPreview from 'app/components/search_preview';
+import StatusBar from 'app/components/status_bar';
+import {ViewTypes} from 'app/constants';
+import {preventDoubleTap} from 'app/utils/tap';
+import {changeOpacity, makeStyleSheetFromTheme} from 'app/utils/theme';
+
+const SECTION_HEIGHT = 20;
+const RECENT_LABEL_HEIGHT = 42;
+const RECENT_SEPARATOR_HEIGHT = 3;
+const POSTS_PER_PAGE = Posts.POST_CHUNK_SIZE / 2;
+const SEARCHING = 'searching';
+
+class Search extends Component {
+    static propTypes = {
+        actions: PropTypes.shape({
+            clearSearch: PropTypes.func.isRequired,
+            getPostsAfter: PropTypes.func.isRequired,
+            getPostsBefore: PropTypes.func.isRequired,
+            getPostThread: PropTypes.func.isRequired,
+            handlePostDraftChanged: PropTypes.func.isRequired,
+            loadThreadIfNecessary: PropTypes.func.isRequired,
+            removeSearchTerms: PropTypes.func.isRequired,
+            searchPosts: PropTypes.func.isRequired,
+            selectPost: PropTypes.func.isRequired
+        }).isRequired,
+        channels: PropTypes.array.isRequired,
+        currentTeamId: PropTypes.string.isRequired,
+        currentChannelId: PropTypes.string.isRequired,
+        intl: intlShape.isRequired,
+        navigator: PropTypes.object,
+        posts: PropTypes.array,
+        recent: PropTypes.array.isRequired,
+        searching: PropTypes.bool,
+        theme: PropTypes.object.isRequired
+    };
+
+    static defaultProps = {
+        posts: []
+    };
+
+    constructor(props) {
+        super(props);
+
+        props.navigator.setOnNavigatorEvent(this.onNavigatorEvent);
+        this.state = {
+            channelName: '',
+            isFocused: true,
+            postId: null,
+            preview: false,
+            value: ''
+        };
+    }
+
+    componentDidMount() {
+        this.refs.search_bar.focus();
+    }
+
+    shouldComponentUpdate(nextProps, nextState) {
+        return (
+            this.props.recent !== nextProps.recent ||
+            this.props.posts !== nextProps.posts ||
+            this.state !== nextState
+        );
+    }
+
+    componentDidUpdate() {
+        const {posts, recent} = this.props;
+        const recentLenght = recent.length;
+
+        if (posts.length && !this.state.isFocused) {
+            this.refs.list.getWrapperRef().getListRef().scrollToOffset({
+                animated: true,
+                offset: SECTION_HEIGHT + (recentLenght * RECENT_LABEL_HEIGHT) + ((recentLenght - 1) * RECENT_SEPARATOR_HEIGHT)
+            });
+        }
+    }
+
+    attachAutocomplete = (c) => {
+        this.autocomplete = c;
+    };
+
+    cancelSearch = () => {
+        const {navigator} = this.props;
+        this.handleTextChanged('');
+        navigator.dismissModal({animationType: 'slide-down'});
+    };
+
+    goToThread = (post) => {
```

Comment:
Is there some way to move this into an action? This looks like very similar to the code in ChannelPostList